### PR TITLE
Enable arm64 linux musl builds

### PR DESCRIPTION
**Pull request by jashook**
Note that -clang5.0 is required to be passed.

### CODE REVIEW

**Comment by jashook** [comment]
I believe this is ready to go. We will be using the linux-arm64 RID for pulling packages. For now this should be acceptable, even if there are failures as there are not yet linux-musl-arm64 packages.

**Comment by jashook** [comment]
@sbomer can you review and check if there are any places I missed for the official build? CI seems fine.

**Comment by jashook** [comment]
Note that this will enable linux arm64 musl builds for official build and ci, but no testing.

File: eng/platform-matrix.yml [hunk 1]
```diff
@@ -79,6 +79,27 @@ jobs:
...
+- template: ${{ parameters.jobTemplate }}
+  parameters:
+    buildConfig: ${{ parameters.buildConfig }}
+    archType: arm64
+    osGroup: Linux
+    osIdentifier: Linux_musl
+    containerName: musl_arm64_build_image
+    helixQueues:
+      ${{ if eq(variables['System.TeamProject'], 'public') }}:
+        # TODO: there are no Alpine arm64 queues https://github.com/dotnet/core-eng/issues/5206
```
Comment (echesakovMSFT) [inline eng/platform-matrix.yml]: I think this is not true. According to 
https://helix.dot.net/api/2018-03-14/info/queues there are 
**Alpine.3.Arm64** and **Alpine.3.Arm64.Open**

Can you please add them here?

Comment (jashook) [reply eng/platform-matrix.yml]: @ulisesh what docker image is this queue running.

Comment (jashook) [reply eng/platform-matrix.yml]: @echesakovMSFT after some more thought, getting clean runs I believe should be out of scope for this change. It currently blocks up-stack components from building. I will enable testing in a separate PR unless there is strong objection.

Comment (echesakovMSFT) [reply eng/platform-matrix.yml]: I am fine with merging this as is and opening another PR to target testing side.

**Comment by jashook** [comment]
Merging and ignoring failures. coreclr-ci had a timeout and osx had a disconnect.

**Pull request merged**